core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -48,8 +48,10 @@ object TransactionCoordinator {
       config.transactionTransactionsExpiredTransactionCleanupIntervalMs)
 
     val producerIdManager = new ProducerIdManager(config.brokerId, zkUtils)
+    // we need to enable reaper thread still even with infinite delays for txn marker purgatory, since we
+    // are still relying on the thread to remove already completed tasks as each task is watched on two lists: txn id and partition id
+    val txnMarkerPurgatory = DelayedOperationPurgatory[DelayedTxnMarker]("txn-marker-purgatory", config.brokerId)
```

Comment:
I meant if we can just get rid of reaperEnabled in class DelayedOperationPurgatory.